Add aboutMe profile character count

nabramow opened the pull request:
<!---
Please describe the pull request below.
If it closes an issue, make sure to write "closes #1234"
If there is an issue but it isn't completely closed, still refer to the issue number, eg. "part of #1234"
--->

User's aren't realizing the reason they can't send messages, etc. is because they need a longer profile. This PR adds a warning text on the EditProfile page aboutMe section if the user is under the limit and same with the missing profile photo:

![Screenshot 2024-12-22 at 7 42 54 PM](https://github.com/user-attachments/assets/95ad6acf-4997-46db-ba98-98e74bab45b7)

![Screenshot 2024-12-22 at 8 08 19 PM](https://github.com/user-attachments/assets/625291cb-afec-42a1-85a4-4849b114c870)

![Screenshot 2024-12-22 at 8 09 46 PM](https://github.com/user-attachments/assets/e5ee5dcb-1c1e-4a85-9576-12a4ac2b1057)

<!---
Checklists - you can remove one that is not applicable (ie. remove backend checklist if you only worked on the web frontend)
If you need help with any of these, please ask :)
--->
**Web frontend checklist**
- [ x ] Formatted my code with `yarn format`
- [ x ] There are no warnings from `yarn lint --fix`
- [ x ] There are no console warnings when running the app
- [ x ] All tests pass
- [ x ] Clicked around my changes running locally and it works
- [ x ] Checked Desktop, Mobile and Tablet screen sizes

<!---
Remember to request review from couchers-org/web, couchers-org/backend or an individual.
Once your code is approved, remember to merge it if you have write access
--->

nabramow committed:
Add aboutMe profile character count

nabramow committed:
Clean up

jesseallhands commented:
This looks to be working as expected and can be merged.

I have a feeling it will be overlooked, especially on mobile.  

Could we add a warning when the user clicks "save" if the character count is still below the limit?

It could say something like "Your "About Me" section is still too empty. Save anyway?" [Yes] [Cancel] and if they click cancel, it can bounce them up to the "About me" section so they can continue editing.

Obviously this would have to be a new issue. Thoughts?

nabramow committed:
Add dialog

aapeliv reviewed (commented):
Tested and seems to be working well.

I suggest using the dialog from #5367 for the helper text. Also I think maybe instead of "Cancel" it's better to say "Continue editing"

aapeliv commented on app/web/features/profile/edit/EditProfile.tsx:
```diff
@@ -134,13 +143,26 @@ export default function EditProfileForm() {
 
+  const handleSubmitButtonClick = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (aboutMeField?.length < 150) {
```
Maybe we can move 150 into a constant. (It's something we've changed before too.)

aapeliv commented on app/web/features/profile/locales/en.json:
```diff
@@ -34,7 +34,15 @@
   "section_tabs_a11y_label": "tabs for user's details",
+  "incomplete_dialog": {
+    "title": "Profile Incomplete",
+    "message": "Your \"Who I Am\" section is still too empty. You won't be able to send requests, messages or create events until this section is at least 150 characters long.",
```
"still too short" maybe?

nabramow committed:
Review changes and missing avatarUrl functionality

nabramow committed:
Fix avatarUrl logic

nabramow committed:
Test fixes

nabramow commented:
@aapeliv @jesseallhands Okay lots of big changes here. I added the alert for the profile photo as well, added a dialog when they click submit that conditionally lists what's missing, stole some of Aapeli's design from his other PR, and updated and added tests.

One thing I wasn't able to test super well - I already have a profile photo and there is no delete option. I tested the UI for that but didn't test it with a real missing photo.

Anyway it's ready again!

aapeliv reviewed (approved):
Tested and code looks good!

aapeliv commented:
(Also I did test without a picture.)
